clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -34,19 +36,22 @@
  *
  */
 final class ClusterConnectionStates {
-    private final long reconnectBackoffInitMs;
-    private final long reconnectBackoffMaxMs;
-    private final static int RECONNECT_BACKOFF_EXP_BASE = 2;
-    private final double reconnectBackoffMaxExp;
     private final Map<String, NodeConnectionState> nodeState;
     private final Logger log;
+    private Set<String> connectingNodes;
+    private GeometricProgression reconnectBackoff;
+    private GeometricProgression connectionSetupTimeout;
 
-    public ClusterConnectionStates(long reconnectBackoffMs, long reconnectBackoffMaxMs, LogContext logContext) {
+    public ClusterConnectionStates(long reconnectBackoffMs, long reconnectBackoffMaxMs,
+                                   long connectionSetupTimeoutMs, long connectionSetupTimeoutMaxMs,
+                                   LogContext logContext) {
         this.log = logContext.logger(ClusterConnectionStates.class);
-        this.reconnectBackoffInitMs = reconnectBackoffMs;
-        this.reconnectBackoffMaxMs = reconnectBackoffMaxMs;
-        this.reconnectBackoffMaxExp = Math.log(this.reconnectBackoffMaxMs / (double) Math.max(reconnectBackoffMs, 1)) / Math.log(RECONNECT_BACKOFF_EXP_BASE);
+        this.reconnectBackoff = new GeometricProgression(
+                reconnectBackoffMs, 2, reconnectBackoffMaxMs, 0.2);
```

Comment:
Just to be sure, we are reusing the code, but not changing the behaviour for backoff right?

Reply:
Yes, we are reusing the code.